src/toil_scripts/tools/preprocessing.py

```diff
@@ -131,9 +262,91 @@ def run_gatk_preprocessing(job, cores, bam, bai, ref, ref_dict, fai, phase, mill
+    :param int cores: Maximum number of cores on a worker node
+    :param str bam: Sample BAM FileStoreID
+    :param str bai: Bam Index FileStoreID
+    :param str ref: Reference genome FileStoreID
+    :param str ref_dict: Reference dictionary FileStoreID
+    :param str fai: Reference index FileStoreID
+    :param str phase: Phase VCF FileStoreID
+    :param str mills: Mills VCF FileStoreID
+    :param str dbsnp: DBSNP VCF FileStoreID
+    :param int file_size: Approximate input BAM size. Default: 10G
+    :param str xmx: Java memory allocation. Default: 10G
+    :param bool unsafe: If True, runs gatk UNSAFE mode: "-U ALLOW_SEQ_DICT_INCOMPATIBILITY"
+    :return: BAM and BAI FileStoreIDs from Print Reads
+    :rtype: tuple(str, str)
+    """
+    # MarkDuplicates runs best when Xmx <= 10G
+    mdups_memory = min(10737418240, xmx)
+    mdups = job.wrapJobFn(picard_mark_duplicates,
+                          bam, bai,
+                          xmx=mdups_memory,
+                          memory=mdups_memory, cores=cores, disk=file_size)
+
+    realigner_target_disk = PromisedRequirement(lambda x: 2 * x.size + human2bytes('10G'),
+                                                mdups.rv(0))
+    realigner_target = job.wrapJobFn(run_realigner_target_creator,
+                                     cores,
+                                     mdups.rv(0), mdups.rv(1),
+                                     ref, ref_dict, fai,
+                                     phase, mills,
+                                     xmx,
+                                     unsafe=unsafe,
+                                     memory=xmx, disk=realigner_target_disk, cores=cores)
+
+    indel_realign_disk = PromisedRequirement(lambda x: 2 * x.size + human2bytes('10G'),
+                                             mdups.rv(0))
+    indel_realign = job.wrapJobFn(run_indel_realignment,
+                                  realigner_target.rv(),
+                                  bam, bai,
+                                  ref, ref_dict, fai,
+                                  phase, mills,
+                                  xmx,
+                                  unsafe=unsafe,
+                                  memory=xmx, disk=indel_realign_disk, cores=cores)
+
+    base_recal_disk = PromisedRequirement(lambda x: 2 * x.size + human2bytes('10G'),
+                                          mdups.rv(0))
+    base_recal = job.wrapJobFn(run_base_recalibration,
+                               cores,
+                               indel_realign.rv(0),
+                               indel_realign.rv(1),
+                               ref, ref_dict, fai,
+                               dbsnp, mills,
+                               xmx,
+                               unsafe=unsafe,
+                               memory=xmx, disk=base_recal_disk, cores=cores)
+
+    recalibrate_reads_disk = PromisedRequirement(lambda x: 2 * x.size + human2bytes('10G'),
```

Comment:
Where ever a constant value (e.g., the '8G' in this line) comes in, you need to document where this constant comes from.